inspect/list_entities/src/index.ts

```diff
@@ -0,0 +1,116 @@
+
+    // Create a QueryContext to be reused for all queries
+    // Limits the queries to only look at the structure
+    const ctx = new QueryContext(struct)
+
+    
+    // ==== Number of Waters ====
+    // Created a query to select all residues that are water
+    // but only select 1 atom per water (ensuring a Singleton selection)
+    const waterQuery = Queries.generators.atoms({
+        'entityTest': ctx => StructureProperties.entity.type(ctx.element) === 'water',
+        'atomTest': ctx => StructureProperties.atom.type_symbol(ctx.element) === 'O'
+    })
+    // Can cast as a Singleton selection since we are only selecting 1 atom per water
+    const waterSelection = waterQuery(ctx) as StructureSelection.Singletons;
+    const numWaters = waterSelection.structure.atomicResidueCount;
+
+
+    // ==== Covalent ligand names and residue code ====
+    // Create a query expression for all ligands connected to the protein
+    const covalentLigandExp = MS.struct.filter.isConnectedTo({
+        0: StructureSelectionQueries.ligand.expression,         // All ligands
+        target: StructureSelectionQueries.protein.expression,   // All protein atoms
+        'bond-test': true  // Only atoms covalently bound to the protein
+    })
+    // Query the atoms with the context to get a StructureSelection
+    const covLigQuery = StructureSelectionQuery('only-covalent-ligands', covalentLigandExp).query;
+    const covLigSelection = covLigQuery(ctx);
+    // Assume ligands in structure have >1 atoms.
+    // Therefore, the StructureSelection must be a Sequence
+    const covLigStructures = (covLigSelection as StructureSelection.Sequence).structures;
+    // Retrieve each ligand name and residue code
+    const covLigNames: string[] = [];
+    const covLigRes: string[] = [];
+    covLigStructures.forEach(s => s.units.map(u => {
+        // Create a location for the first element of the ligand
+        // to retrieve structure properties
+        const location = Location.create(s, u, u.elements[0])
+        // Return the ligand name property for the ligand
+        const name = StructureProperties.entity.pdbx_description(location).join('|')
+        covLigNames.push(name);
+        // Return the residue code for the ligand
+        const res = StructureProperties.atom.label_comp_id(location)
+        covLigRes.push(res);
+    }))
+    
+
+    // ==== Number of AltLoc positions ====
+    const altLocQuery = Queries.generators.atoms({
+        // Any atom with a non '' alt_id
+        'atomTest': ctx => !!StructureProperties.atom.label_alt_id(ctx.element),
+    });
+    // Can only select 1 atom at a time, must be a Singleton
+    const altLocSelection = altLocQuery(ctx) as StructureSelection.Singletons;
+    const numAltLocs = altLocSelection.structure.elementCount;
+
+    
+    // ==== Polymer ASYM Unit name and chain ====
+    const polymerSelection = StructureSelectionQueries.polymer.query(ctx)
+    // Assume more than 1 atom in the polymer entity
```

Comment:
"polymer query groups selected atoms by entity. Assume it creates a Sequence StructureSelection"